pkg/sql/model.go

```diff
@@ -183,3 +198,86 @@ func readGob(filePath string, object interface{}) error {
 	}
 	return nil
 }
+
+// createModelZooTable create the table "sqlflow.trained_models" to save model
+// metas the saved model URI.
+func createModelZooTable(db *DB) error {
+	createDBSQL := fmt.Sprintf("CREATE DATABASE IF NOT EXISTS %s;", modelZooDB)
+	_, err := db.Exec(createDBSQL)
+	if err != nil {
+		return err
+	}
+	// schema design: https://github.com/sql-machine-learning/sqlflow/blob/a98218ef8bee57e2a45357d7ee5721e1c6dfeb35/doc/design/model_zoo.md#model-zoo-data-schema
+	// NOTE(typhoonzero): submitter program size may exceed TEXT size(64KB)
+	// NOTE(typhoonzero): train_ir_pb contains all information how the submitter program is generated, so not saving submitter program now
+	// NOTE(typhoonzero): model_uri can be:
+	//    1. file:///path/to/model/dir
+	//    2. db.table
+	//    3. oss://path/to/oss
+	createTableSQL := fmt.Sprintf(`CREATE TABLE IF NOT EXISTS %s (
+model_id VARCHAR(255),
+author VARCHAR(255),
+model_image VARCHAR(255),
+model_def VARCHAR(255),
+train_ir_pb TEXT,
+model_uri VARCHAR(255)
+);`, modelZooTable)
+	_, err = db.Exec(createTableSQL)
+	return err
+}
+
+func getTrainedModelParts(into string) (string, string, error) {
+	if strings.ContainsRune(into, '.') {
+		parts := strings.Split(into, ".")
+		if len(parts) == 2 {
+			return parts[0], parts[1], nil
+		}
+		return "", "", fmt.Errorf("error INTO format, should be like [creator.]modelID, but got %s", into)
+	}
+	return "", into, nil
+}
+
+func dbStringEscape(src string) string {
+	ret := strings.ReplaceAll(src, "\"", "\\\"")
+	return strings.ReplaceAll(ret, "'", "\\'")
+}
+
+func addTrainedModelsRecord(db *DB, trainIR *ir.TrainClause, modelURI string, sess *pb.Session) error {
+	fmt.Println("in addTrainedModelsRecord")
+	// NOTE(typhoonzero): creator can be empty, if so, the model file is saved into current database
+	// FIXME(typhoonzero): or maybe the into format should be like "creator/modelID"
+	creator, modelID, err := getTrainedModelParts(trainIR.Into)
+	q := fmt.Sprintf("SELECT * FROM %s WHERE model_id='%s'", modelZooTable, modelID)
+	fmt.Println(q)
+	res, err := db.Query(q)
+	if err != nil {
+		return err
+	}
+	defer res.Close()
+	isInsert := false
+	if !res.Next() {
+		isInsert = true
```

Comment:
Maybe add a test examing the `isInsert` logic.

Reply:
`CaseTrainFeatureDerevation` has two train SQL which have the same `INTO` clause, this is testing this if branch. Adding comments in that test case.